src/oid4vci/mod.rs

```diff
@@ -0,0 +1,366 @@
+mod error;
+mod http_client;
+mod metadata;
+mod proof_of_possession;
+mod session;
+mod wrapper;
+
+pub use error::*;
+pub use http_client::*;
+pub use metadata::*;
+pub use proof_of_possession::*;
+pub use session::*;
+pub use wrapper::*;
+
+use std::sync::Arc;
+
+use either::Either;
+use oid4vci::{
+    core::{
+        client,
+        metadata::CredentialIssuerMetadata as ICredentialIssuerMetadata,
+        profiles::{
+            self,
+            w3c::{CredentialDefinition, CredentialDefinitionLD},
+            CoreProfilesConfiguration, CoreProfilesOffer, CoreProfilesResponse,
+        },
+    },
+    credential::ResponseEnum,
+    credential_offer::CredentialOfferParameters,
+    metadata::AuthorizationMetadata,
+    openidconnect::{AuthorizationCode, ClientId, IssuerUrl, OAuth2TokenResponse, RedirectUrl},
+    profiles::CredentialConfigurationProfile,
+    proof_of_possession::Proof,
+};
+use ssi::{
+    claims::{
+        jwt::ToDecodedJWT, vc::v1::data_integrity::any_credential_from_json_str,
+        VerificationParameters,
+    },
+    dids::{AnyDidMethod, DIDResolver},
+};
+
+use crate::vdc_collection::CredentialFormat;
+
+#[uniffi::export]
+pub async fn oid4vci_initiate_with_offer(
+    credential_offer: String,
```

Comment:
What's the format of this string? The code implies that it's JSON, but a credential offer is parsed from a URI. Could we just pass the URI here instead?